process_csv.php

```diff
@@ -0,0 +1,218 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+namespace enrol_oneroster;
+
+use enrol_oneroster\classes\form\oneroster_org_selection_form;
+use enrol_oneroster\classes\form\oneroster_csv_form;
+use enrol_oneroster\classes\local\csv_client_helper;
+
+require_once('../../config.php');
+require_once(__DIR__ . '/classes/form/oneroster_csv_form.php');
+require_once(__DIR__ . '/classes/form/oneroster_org_selection_form.php');
+require_once(__DIR__ . '/classes/local/csv_client_helper.php');
+require_once(__DIR__ . '/classes/local/csv_client_const_helper.php');
+require_once($CFG->libdir . '/adminlib.php');
+
+/**
+ * One Roster Client
+ *
+ * @package    enrol_oneroster
+ * @copyright  Gustavo Amorim De Almeida, Ruben Cooper, Josh Bateson, Brayden Porter
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ * 
+ */
+
+admin_externalpage_setup('enrol_oneroster_csv_upload');
+
+$PAGE->set_url(get_string('set_url', 'enrol_oneroster'));
+$PAGE->set_context(\context_system::instance());
```

Comment:
set_context is already handled by admin_externalpage_setup if i  remember correctly